[22기 이경민]  로그인기능 , 댓글 컴포넌트화, props로 데이터 전달,비동기 문제 해결

glorious-min opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
댓글 컴포넌트화 + props로 데이터 전달
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)


## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)

- 메인에서 props 에 key 가 각각 다른 i로 제대로 지정된것인지 궁금합니다. 
-메인과 댓글 컴포넌트간의 state 전달이 잘 된것인지 궁금합니다 


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

glorious-min committed:
경민 임시저장

glorious-min committed:
merge branch 'master' into feature/gyeongminLee

glorious-min committed:
test1

glorious-min committed:
Add : Mission 1 - 사용자 입력 데이터 저장 기능 구현

glorious-min committed:
Add : Mission 2 - 로그인 버튼 활성화 기능 구현

glorious-min committed:
Add : Mission 3 - 댓글 기능 구현

glorious-min committed:
Add : Mission 4 - 댓글 컴포넌트화, props로 데이터 전달 구현

glorious-min committed:
수정중-이벤트 수정 후 댓글입력 불가능해짐(피드백 대상 아님)

glorious-min committed:
[Assignment1~4] 로그인기능 완료 , 댓글 컴포넌트화 + props로 데이터 전달

glorious-min committed:
signment1~4] 로그인기능 완료 , 댓글 컴포넌트화 + props로 데이터
전달(비동기 문제 해결)

93-0312 commented on .eslintrc:
```diff
@@ -1,3 +1,7 @@
\ No newline at end of file
+  "extends": ["react-app", "plugin:prettier/recommended"],
+  "rules": {
+    "linebreak-style": 0, // "off" 옵션을 주면 규칙이 아예 작동하지 않음
+    "prettier/prettier": ["error", { "endOfLine": "auto" }]
```
윈도우 세팅 수고하셨습니다

younuk23 replied:
주석은 지워주세요, 그리고 세팅관련은 세팅관련 브랜치 따로 파서 적용한 뒤에 다시 머지받도록하는 프로세스로 진행해주세요
이렇게 개인 작업브랜치에서 세팅관련된 내용을 건드리는건 좋지 않습니다!

93-0312 commented on .vscode/settings.json:
```diff
@@ -7,5 +7,5 @@
   "javascript.format.enable": false,
   "eslint.alwaysShowStatus": true,
-  "files.autoSave": "onFocusChange"
+  "files.autoSave": "off"
```
이건 뭔가요?

93-0312 commented on public/index.html:
```diff
@@ -3,6 +3,10 @@
   <head>
     <meta charset="utf-8" />
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <script
+      src="https://kit.fontawesome.com/10fb066d8f.js"
+      crossorigin="anonymous"
+    ></script>
```
폰트어썸 스크립트 추가하신이유가 뭔가요?? 뭔가 작동이 안됬다던가??

public 안에 index.html은 초기세팅할때 정해놓은 부분이라 무언가 달라지면 같이 작업하는분들도 마스터를 다시 pull해야 하거나 해야 하지 않나??싶습니다~~

93-0312 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,84 @@
 import React, { Component } from 'react';
-
+import './Login.scss';
 class Login extends Component {
```
사이에 엔터 한칸 있는건가요? 없으면 한칸 추가해주는게 보기 좋을 것 같아요~

younuk23 replied:
👍

93-0312 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,84 @@
+    if (event.target.type === 'text') {
+      this.setState({
+        idValue: event.target.value,
+      }); // 실행이 완료될때까지 기다리지 ㅇ낳는다 - setState -> 비동기 아이디 밸류 달라지기 전에 아래 함수 실행함
```
필요없는 주석은 삭제해주시고, 혹시 필요한 내용이라면 그냥 두셔도 좋을 것 같아요~

93-0312 commented on src/pages/gyeongminLee/Main/Comment/Comment.js:
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+class Comment extends React.Component {
+  render() {
+    //console.log(this.props.comment);
```
주석 지워주세요

93-0312 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,228 @@
+      });
+    }
+  };
+  // submitButtonDisabled = event => {
+  //   if (event.target.type === 'button') {
+  //     this.setState({
+  //       submitButtonDisabled: event.target.disabled,
+  //     });
+  //   }
+  // };
```
이런 안쓰는건 지우는게 좋지않을까 싶습니다

93-0312 reviewed (commented):
1차 피어리뷰입니다. 화이팅~~~~

eeesssooo commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,84 @@
+    };
+  }
+
+  controlValue = event => {
+    if (event.target.type === 'text') {
+      this.setState({
+        idValue: event.target.value,
+      }); // 실행이 완료될때까지 기다리지 ㅇ낳는다 - setState -> 비동기 아이디 밸류 달라지기 전에 아래 함수 실행함
+    } else if (event.target.type === 'password') {
+      this.setState({
+        passwordValue: event.target.value,
+      });
+    }
+  };
```
이렇게 같은 함수로 묶어줄 수 있네요 !

eeesssooo commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,228 @@
+    }
+  };
+  activateSubmitButton = e => {
+    if (this.state.newCommentValue.length > 0) {
```
배열의 길이가 0일 때도 포함 시켜주면 댓글 입력할때 첫번째 글자에서부터 '게시' 버튼 컬러가 바뀌지 않을까요 ?

eeesssooo commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,228 @@
+                  src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/2wCEAAkGBwgHBgkIBwgKCgkLDRYPDQwMDRsUFRAWIB0iIiAdHx8kKDQsJCYxJx8fLT0tMTU3Ojo6Iys/RD84QzQ5OjcBCgoKDQwNGg8PGjclHyU3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3Nzc3N//AABEIAGQAZAMBIgACEQEDEQH/xAAbAAACAwEBAQAAAAAAAAAAAAAEBQADBgECB//EADQQAAIBAwIDBgQFBQEBAAAAAAECAwAEERIhBTFBEyJRYXGBBhSRsTJCocHRFSNS4fBiB//EABoBAAICA ...
+                />
+                <div id="feedProfileAccount">
+                  {' '}
```
{' '} 이게 무엇이죠 ? !

93-0312 replied:
뭐죠?

younuk23 replied:
formatting 되면서 생긴 것 같은데 불필요한 것들 다 삭제해주세요!

eeesssooo commented on src/pages/gyeongminLee/Main/Main.scss:
```diff
@@ -0,0 +1,220 @@
+  width: 100%;
+  height: 100%;
+  padding-left: 10%;
+  background-color: #fafafa;
```
함께 만든 variables.scss 에 있는 변수를 사용해보는건 어때요 경민님 ?ㅎㅎ

eeesssooo reviewed (commented):
경민님 위스타그램 구현하시느라 고생하셨어요 ! :) Nav도, icon도 잘 작동되어 다행이에요 ~ 저의 코드 리뷰가 경민님에게 도움이 되었길 바래요 💌

glorious-min committed:
로그인 페이지 비동기 문제 해결 + 리팩토링 진행중

glorious-min committed:
불필요한 state 줄이기 전 , 댓글쓰기 작동 안 하는 상태

93-0312 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,210 @@
+        {
+          newCommentValue: event.target.value,
+        },
+        () => this.activateSubmitButton(event) //(중요!!!!!!!)콜백함수 - 이것때문에 함수 다시 한번 실행되어 댓글이 두개씩 달림!!
+        // 일단 제외하고 제이쓴 기능 구현하기 !!!!!!!!(이후 리팩토링 필요)
```
왜 두번 되는건가요?

93-0312 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,210 @@
+    }
+  };
+
+  controlKeyPress = event => {
+    // 태그안에 인풋창에서 키가 눌린걸 감지하는 이벤트를 걸어야하는거 아닌가 ?
+    if (event.key === 'Enter') {
+      this.setState({
+        list: this.state.list.concat(
+          //...<=stackoverflow //이걸로도 한 번 해보기
+          //num: this.num,
+          //  num : this.num,
+          //newCommentValue: newCommentValue,
+          [this.state.newCommentValue]
+        ),
+        // list: [
+        //   ...this.state,
+        //   {
+        //     id: this.state.length + 1,
+        //     content: this.state.newCommentValue, //아닐것 같은 부분
+        //   },
+        // ],
+        newCommentValue: '', //이거 빼고 한번 해보기
+      });
+      //this.num += 1;
+    }
+  };
```
... 전개 연산자도 사용하기 편한 것 같아요

93-0312 reviewed (commented):
다른 조에 비해서 뭔가 대충 리뷰했나 싶어서 다시 봤는데, 뭘 어떻게 써야 될지 모르겠네요,,

glorious-min committed:
불필요한 state 삭제 전

glorious-min committed:
목업(js) 완료

younuk23 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,151 @@
+  }
+
+  controlValue = event => {
+    if (event.target.type === 'text') {
```
경민님 불필요한 주석은 다 삭제해주세요!
주석때문에 가독성이 너무 떨어집니다!

younuk23 replied:
input에는 name property를 줄 수 있습니다, type으로 어떤 input인지 판단하는것보다 name으로 판단하는게 더 문제의 여지가 적어보이고 적합해보입니다!

younuk23 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,151 @@
+        {
+          idValue: event.target.value,
+        },
+        () => this.controlActivation(event) //정리해야할것 - 기존의 컨트롤 액티베이션 함수를 태그에 넣을 필요없이
```
controlActivation함수 없어진걸로 보이는데 계속 호출하고있네요 에러 발생할 것 같습니다!

younuk23 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,151 @@
+            //응답은 맞는경우도 있고 아닌경우도 있다 - if문으로 나누어 주어야된다
+            this.props.history.push('/main-Gyeongmin');
+            //alert(result.ACCESS_TOKEN);
+            localStorage.setItem('Token', result['ACCESS_TOKEN']);
```
Token -> token
camelCase 사용해주세요!

younuk23 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,151 @@
+    //console.log(JsonResponse['ACCESS_TOKEN']);
+    // const idValid = this.state.idValue.length >= 1; 멘토님의 것
+    //const inputValid =this.state.idValue.includes('@')&&this.state.password.length>=5
+    const idValid = this.state.idValue.includes('@');
+    const passwordValid = this.state.passwordValue.length >= 5;
```
```suggestion
    const idValid = this.state.idValue.includes('@');
    const passwordValid = this.state.passwordValue.length >= 5;
    const isAllValueValid = idValid && passwordValid
    
                  backgroundColor={
              isAllValueValid   === true ? '#B2DFFC' : '#0095F6'
              }
              disabled={isAllValueValid === true ? false : true}
```
모두 유효한지에 대한 정보를 background, disabled에서 두번 사용하는 것 같은데 이런경우 변수화 해서 중복없애서 사용할 수 있겟네요!

younuk23 replied:
boolean은 true or false의 정보이기 때문에 변수명으로 is~~~로 사용하는 컨벤션이 있습니다!

younuk23 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,151 @@
+    return (
+      <>
+        <div className="container">
+          <form id="logo" action="index.html" method="post">
+            <a id="logo">westagram</a>
```
- `id` 보다는 `className` 을 활용해주시기 바랍니다.
- HTML에서  `id` 는 문서 내에서 유일해야 합니다. 이러한 이유로 단일 요소를 다루는 경우 `id`를 사용할 수 있습니다.
- 하지만 정말로 해당 요소에 대한 이벤트나 스타일 핸들링이 프로젝트 전체에서 단 한번만 사용 되는지 충분히 고민해야합니다. 일반적으로 컴포넌트는 재사용 될 수 있기 때문에 같은 요소가 여러번 생성될 수 있습니다. 이러한 이유로 보통의 경우에는 `className` 을 사용합니다.

younuk23 replied:
전반적으로 id를 활용하신 경우가 많아보이는데 모두 className 으로 사용해주세요!

younuk23 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,151 @@
+              onClick={this.goToMain}
+              type="button"
+              id="loginButton"
+              backgroundColor={
+                idValid && passwordValid === true ? '#B2DFFC' : '#0095F6'
+              }
```
style={{backgroundColor:~~~}}로 해야 바뀌지 않을까요? 지금은 아무런 동작도 안 할 것 같습니다!

younuk23 replied:
그리고 인라인 스타일 사용보다는 className의 변화를 통해서 동적인 스타일링 해주세요

younuk23 commented on src/pages/gyeongminLee/Login/Login.js:
```diff
@@ -1,8 +1,151 @@
+              onClick={this.goToMain}
+              type="button"
+              id="loginButton"
+              backgroundColor={
+                idValid && passwordValid === true ? '#B2DFFC' : '#0095F6'
+              }
+              disabled={idValid && passwordValid === true ? false : true}
+            >
```
삼항연산자 안쓰고 boolean의 특성을 이용해서 간결하게 표현 가능해보입니다!
```suggestion
              backgroundColor={
                idValid && passwordValid ? '#B2DFFC' : '#0095F6'
              }
              disabled={!(idValid && passwordValid)}
            >
```

younuk23 commented on src/pages/gyeongminLee/Login/Login.scss:
```diff
@@ -0,0 +1,64 @@
+@font-face {
+  font-family: 'Lobster', cursive;
+  font-style: normal;
+  font-weight: 400;
+  src: url('https://fonts.googleapis.com/css2?family=Lobster&display=swap');
+}
```
이런 부분은 commons.scss에 들어가야 하지 않을까요?

younuk23 commented on src/pages/gyeongminLee/Login/Login.scss:
```diff
@@ -0,0 +1,64 @@
+  position: absolute;
+  width: 400px;
+  height: 400px;
+  margin-left: -200px;
+  margin-top: -200px;
```
absolute 이용해서 가운데 정렬을 맞추기 위해서 사용하신걸까요?
margin-left,top은 width,height가 바뀔 때마다 같이 수정해줘야하기때문에 transform:translast(-50%,-50%)를 이용하시는게 좋아보입니다!

younuk23 commented on src/pages/gyeongminLee/Main/Comment/Comment.js:
```diff
@@ -0,0 +1,53 @@
+  render() {
+    //console.log(this.props.comment);
+    return (
+      <>
+        <li>
+          <span>{this.props.comment}</span>
+          <button>x</button>
+        </li>
+      </>
```
이미 단일태그인 li로 되어있기 때문에 Fragment 사용은 불필요해보입니다!

younuk23 commented on src/pages/gyeongminLee/Main/Comment/commentData.json:
```diff
@@ -0,0 +1,17 @@
+  },
+  {
+    "id": 2,
+
```
불필요한 공백같아보입니다

younuk23 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,210 @@
+  constructor() {
+    super();
+    this.state = {
+      submitButtonColor: 'none',
```
submitButtonColor는 state로 관리할 필요 없습니다 -> 다른 state로부터 계산 가능한 값이기에

younuk23 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,210 @@
+        {
+          newCommentValue: event.target.value,
+        },
+        () => this.activateSubmitButton(event) //(중요!!!!!!!)콜백함수 - 이것때문에 함수 다시 한번 실행되어 댓글이 두개씩 달림!!
```
필요없어진 함수같은데 삭제해주세요

younuk23 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,210 @@
+      <>
         <Nav />
-      </div>
+        <main className="mainPageMain">
```
최상단 className은 컴포넌트와 동일하게 지어주시는게 좋아보입니다!
mainPageMain은 네이밍이 너무 이상하네요

younuk23 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,210 @@
+    }
+    this.controlKeyPress(event);
+  };
+
```
경민님 코드 전반적으로 불필요한 주석이 너무 많아서 가독성을 해치고 있습니다!
불필요한 주석들은 다 삭제해주세요!

younuk23 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,210 @@
+              </div>
+
+              <div className="feedCommentsListContainer">
+                {this.state.list.map((content, id) => {
+                  return <Comment comment={content} key={id} />;
+                })}
```
map의 두번째 인자는 index가 들어옵니다 이걸 id라고 이름을 지으면 보는사람이 헷갈릴 수 있겠네요 보통 많이사용하는 index의  축약어인 idx라고 쓰시거나 index라고 전체 다 적어주는게 좋아보입니다!
그리고 key값을 가장 먼저 적어주세요 
```suggestion
                {this.state.list.map((content, idx) => {
                  return <Comment key={idx} comment={content}  />;
                })}
```
그리고 index는 배열의 중간요소가 삭제될 때 변할 수 있기 때문에 유니크하고 변하지 않는 값이여야하는 key값으로는 적절치 않습니다!
다른 값을이용해서 key props부여해주세요

younuk23 commented on src/pages/gyeongminLee/Main/Main.js:
```diff
@@ -1,15 +1,210 @@
+                />
+                <button
+                  //onChange={this.controlButton}
+                  style={{ backgroundColor: this.state.submitButtonColor }}
```
inline style 지양해주세요

younuk23 commented on src/pages/gyeongminLee/Main/Main.scss:
```diff
@@ -0,0 +1,220 @@
+  border-bottom: 1px #dbdbdb solid;
+  padding-left: 80px;
+  padding-right: 80px;
+  nav {
```
다른 셀렉터들끼리는 한줄 공백 주세요!

younuk23 replied:
태그 선택자는 다른 요소들에 영향을 받을, 또 영향을 미칠 수 있는 여지가 너무 많기 때문에 사용을 지양해주셔야합니다,
스타일링이 필요할 경우 className을 부여하셔서 사용해주세요!

younuk23 commented on src/pages/gyeongminLee/Main/Main.scss:
```diff
@@ -0,0 +1,220 @@
+  }
+
+  div.main-right {
+    float: left;
```
float은 사용하지말아주세요!
유지보수하기 힘든 스타일 속성이기에 사용을 지양하고 있습니다!
flex 속성이 등장하고나서부터는 사용할 이유가 없습니다!

younuk23 commented on src/pages/gyeongminLee/Main/Main.scss:
```diff
@@ -0,0 +1,220 @@
+
+  div.main-right {
+    float: left;
+    height: auto;
```
height:auto는 기본값이라서 따로 적용안하셔도 무방해보입니다!

younuk23 commented on src/pages/gyeongminLee/Main/Main.scss:
```diff
@@ -0,0 +1,220 @@
+.topheader {
```
topheader는 어디에 스타일을 주고 있는건가요?  Main.scss에는 해당 className이 보이지않네요
컴포넌트의 scss파일에서는 해당 컴포넌트에 해당하는 부분만 스타일링 해주세요!

yeonuk-hwang reviewed (commented):
메인에서 props 에 key 가 각각 다른 i로 제대로 지정된것인지 궁금합니다.
-메인과 댓글 컴포넌트간의 state 전달이 잘 된것인지 궁금합니다
-> 의문을 가지고 확인해보려하는 것 너무 좋네요, 다만 궁금하시면 console 찍어서 확인하셔야합니다!!
- PR template 작업하신 내용 잘 정리해서 작성해주세요 지금은 제대로 기입되지 않은 것 같습니다!

수고하셨습니다 리뷰 확인 및 반영해주세요~!

glorious-min committed:
Mock up 댓글 진행중

glorious-min committed:
메인페이지 버튼기능 리팩토링 / 댓글입력안됨

glorious-min committed:
로그인, 메인 페이지 불필요한 함수제거, 댓글 Json 이용 데이터 입력

glorious-min committed:
라이프 사이클 확인 위해 콘솔 찍으며 확인한 것

glorious-min committed:
주석지운 버전 - 댓글 데이터 전달까지 완성
i Please enter the commit message for your changes. Lines starting

glorious-min committed:
불필요 주석 모두 삭제 댓글 데이터 fetch 사용

glorious-min committed:
피드 게시물 하나 게시 성공
mmPlease enter the commit message for your changes. Lines starting

glorious-min committed:
게시물 피드 성공

glorious-min committed:
게시물 피드 구현 +피드 key 부여
li key={comment.id}># Please enter the commit message for your changes. Lines starting

glorious-min committed:
Warning: Comment: `key` is not a prop.
